server/src/views/frame.js

```diff
@@ -222,6 +222,7 @@ class Frame extends React.Component {
 
   clickFullPageButton(e) {
     e.preventDefault();
+    history.pushState({state: "frame"}, "", "#frame");
```

Comment:
It moves code around, but I don't think it's more code.  And it means no matter how you get the hash in your URL, you'll get consistent behavior.